src/OrleansRuntime/Messaging/OutboundMessageQueue.cs

```diff
@@ -125,6 +127,22 @@ public void SendMessage(Message msg)
                     return;
                 }
 
+                // check for simulation of lost messages
+                if (Silo.CurrentSilo.TestHookup.SimulatedMessageLoss != null)
```

Comment:
Can you elaborate?

(e.g. what is wrong with it, suggest better ways to make this happen?)